File: frontend/src/api/knowledge.ts
import api from './index'

export interface KnowledgeCategory {
  id: number
  name: string
  icon: string
  color: string
  sort_order: number
  card_count: number
}

export interface KnowledgeCard {
  id: number
  category_id: number
  category_name: string
  title: string
  content: string
  tags: string
  created_at: string
  updated_at: string
}

export const knowledgeApi = {
  // 分类
  listCategories: () => api.get<any, KnowledgeCategory[]>('/api/knowledge/categories'),
  createCategory: (data: Partial<KnowledgeCategory>) => api.post('/api/knowledge/categories', data),
  deleteCategory: (id: number) => api.delete(`/api/knowledge/categories/${id}`),

  // 卡片
  listCards: (params?: { category_id?: number; keyword?: string }) =>
    api.get<any, KnowledgeCard[]>('/api/knowledge/cards', { params }),
  getCard: (id: number) => api.get<any, KnowledgeCard>(`/api/knowledge/cards/${id}`),
  createCard: (data: Partial<KnowledgeCard>) => api.post('/api/knowledge/cards', data),
  updateCard: (id: number, data: Partial<KnowledgeCard>) => api.put(`/api/knowledge/cards/${id}`, data),
  deleteCard: (id: number) => api.delete(`/api/knowledge/cards/${id}`),

  // 随机推送
  randomCard: () => api.get<any, KnowledgeCard>('/api/knowledge/random'),

  // 统计
  getStats: () => api.get<any, { total_categories: number; total_cards: number }>('/api/knowledge/stats'),
}

File: backend/app/schemas/knowledge.py
"""Knowledge module schemas."""

from pydantic import BaseModel
from datetime import datetime
from typing import Optional


# ── Category schemas ──────────────────────────────────────────


class CategoryCreate(BaseModel):
    name: str
    icon: str = "📚"
    color: str = "#409EFF"
    sort_order: int = 0


class CategoryResponse(BaseModel):
    id: int
    name: str
    icon: str
    color: str
    sort_order: int
    card_count: int = 0

    class Config:
        from_attributes = True


# ── Card schemas ──────────────────────────────────────────────


class CardCreate(BaseModel):
    category_id: int
    title: str
    content: str
    tags: str = ""


class CardUpdate(BaseModel):
    title: Optional[str] = None
    content: Optional[str] = None
    tags: Optional[str] = None
    category_id: Optional[int] = None


class CardResponse(BaseModel):
    id: int
    category_id: int
    category_name: str = ""
    title: str
    content: str
    tags: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


# ── Stats schema ──────────────────────────────────────────────


class KnowledgeStatsResponse(BaseModel):
    total_cards: int
    total_categories: int
